### ISSUE TITLE
Dapr watchdog logs "Failed to watch *v1.Pod: unknown (get pods)"

### ISSUE THREAD

**Issue by alecor191** [dapr/dapr#6526]
## In what area(s)?

> /area operator

## What version of Dapr?

> 1.10

## Expected Behavior

No error logs in `dapr-operator` logs when enabling watchdog.

## Actual Behavior

After enabling the Watchdog, after startup of `dapr-operator`, I get ~4 of the below logs and then every minute or so a few more (maybe related to watchdog poll interval set to 1min in my case):

```shell
2023-06-16T08:38:40+02:00 E0616 06:38:40.757688       1 reflector.go:140] pkg/mod/k8s.io/client-go@v0.26.1/tools/cache/reflector.go:169: Failed to watch *v1.Pod: unknown (get pods)
2023-06-16T08:39:17+02:00 E0616 06:39:17.176972       1 reflector.go:140] pkg/mod/k8s.io/client-go@v0.26.1/tools/cache/reflector.go:169: Failed to watch *v1.Pod: unknown (get pods)
```

## Steps to Reproduce the Problem

Just enable the Watchdog, let it run for a few minutes and logs above show up

I tested it on an AKS cluster v1.25.6, as well as Docker Desktop K8S locally. Same result.

I found that the above logs also show up in some other issues:
- https://github.com/dapr/dapr/issues/6263
- https://github.com/dapr/dapr/issues/6241

BTW what's the easiest way to test if Watchdog works (i.e. startup app pods with Dapr annotation but without having the sidecar injected, to observe watchdog taking action and injecting the missing sidecar)? I tried to stop the AKS cluster and restart it, but all my apps start up with Dapr sidecar nicely injected.

**Comment by alecor191** [comment]
**Update**: I was able to test the watchdog as after stopping/starting AKS clusters a couple of times, eventually some pods started up without sidecar injected. Dapr watchdog then nicely kicked-in and deleted affected pods, triggering a restart of the pod.

Hence, the watchdog seems to work fine. However, I still have the issue with the error logs mentioned above getting added on every watchdog check interval.

**Comment by archana-punia** [comment]
can you paste here the permissions you gave for the watchdog?

**Comment by alecor191** [comment]
**TL;DR**: Upgrading to Dapr `1.11.1` solved this issue for me.

### Context
I didn't provide any specific permissions. I just ran the Dapr Helm chart and enabled the watchdog.

I found the following GitHub issue indicating that missing permissions could be the issue: https://github.com/kubernetes/client-go/issues/829

I then updated Dapr from `1.10` to `1.11.1` and the error message disappeared. Taking a closer look, it seems Dapr `1.11.1` added the missing `watch` permission as mentioned in the GitHub issue above:

### Dapr 1.10 (not working)

```
❯ k describe clusterrole dapr-operator-admin
Name:         dapr-operator-admin
...
PolicyRule:
  Resources                                       Non-Resource URLs  Resource Names  Verbs
  ---------                                       -----------------  --------------  -----
  ...
  pods                                            []                 []              [get list delete]
```

### Dapr 1.11.1 (working)

```
❯ k describe clusterrole dapr-operator-admin
Name:         dapr-operator-admin
...
PolicyRule:
  Resources                                       Non-Resource URLs  Resource Names  Verbs
  ---------                                       -----------------  --------------  -----
  ...
  pods                                            []                 []              [get list delete watch]
  ...
```
